File: programmers/4-43165-타겟넘버/junghwan.py
'''
19. 06. 22

DFS 개념 활용
덧셈과 뺄셈을 했을 경우를 더하면서 결과 도출

'''

def solution(numbers, target):
    answer = calc(numbers, target, 0, 0)
    return answer


def calc(numbers, target, _sum, idx):
    
    if idx == len(numbers):
        if _sum == target:
            return 1
        else:
            return 0
    
    return calc(numbers, target, _sum+numbers[idx], idx+1) + calc(numbers, target, _sum-numbers[idx], idx+1)

File: programmers/8-42746-가장큰수/junghwan.js
function solution(numbers) {
    var answer = '';
    var ns = numbers.map(e => e.toString());
    
    ns.sort((a, b) => {
        var sum = a + b;
        var reverseSum = b + a;
        
        if(sum >= reverseSum) {
            return -1;
        }
        
        else {
            return 1;
        }
    });
    
    answer = ns.join('');
    
    if(ns[0] === '0') {
        ns.shift();
    }
    
    let cnt = 0;
    for(let i=0; i<ns.length; i++) {
        if(ns[i] === '0') {
            cnt++;
        }
    }
    
    if(cnt === ns.length) {
        answer = '0';
    }
    
    
    return answer;
}